modules/Microsoft.HealthDataServices/fhirservices/.bicep/nested_roleAssignments.bicep

```diff
@@ -0,0 +1,74 @@
+@sys.description('Required. The IDs of the principals to assign the role to.')
+param principalIds array
+
+@sys.description('Required. The name of the role to assign. If it cannot be found you can specify the role definition ID instead.')
+param roleDefinitionIdOrName string
+
+@sys.description('Required. The resource ID of the resource to apply the role assignment to.')
+param resourceName string
```

Comment:
```suggestion
param resourceId string
```

Reply:
I've had problems using resourceId when the resource is a child resource hence I switched to passing the resourceName and parent name. I can update the code comments and make sure it mentions resourceName vs ID.

```
"code": "InvalidTemplate",
			"message": "Deployment template validation failed: 'The template resource '[guid(resourceId('Microsoft.HealthcareApis/workspaces/fhirservices', split(last(split(parameters('resourceId'), '/')), '/')[0], split(last(split(parameters('resourceId'), '/')), '/')[1]), parameters('principalIds')[copyIndex()], parameters('roleDefinitionIdOrName'))]' at line '1' and column '5010' is not valid: The language expression property array index '1' is out of bounds.. Please see https://aka.ms/arm-template-expressions for usage details.'."
```